### PR TITLE
Add voting support (motions)

### PR DESCRIPTION
**Pull request by willm30**
This PR adds support for the `MotionVoteSubmitted` event (i.e. motion voting).

Test with: https://github.com/JoinColony/colonyCDapp/pull/380

### CODE REVIEW

**Review by rdig — APPROVED** [review]
All seems good, and as far as I can tell it also supports the user changing their vote, prior to the reveal period

File: src/handlers/motions/motionVoteSubmitted/helpers.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    ];
+  }
+
+  return voterRecord.map((record) => {
+    const { address } = record;
+    if (address === voterAddress) {
+      const { voteCount } = currentVoterRecord;
+      const updatedRecord: VoterRecord = {
+        ...currentVoterRecord,
+        voteCount: BigNumber.from(voteCount).add(1).toString(),
```
[the remark this answers is not in the document]
Comment (willm30) [reply src/handlers/motions/motionVoteSubmitted/helpers.ts:28]: It's a counter for how many times a specific user voted. I'm using it to detect in the front end if a user changed their vote.

File: src/handlers/motions/motionVoteSubmitted/index.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  const votingClient = await getVotingClient(colonyAddress);
+  const { chainId } = await votingClient.provider.getNetwork();
+  const motionDatabaseId = getMotionDatabaseId(
+    chainId,
+    votingClient.address,
+    motionId,
+  );
+  const votedMotion = await getMotionFromDB(colonyAddress, motionDatabaseId);
+
+  if (votedMotion) {
```
Comment (jakubcolony) [inline src/handlers/motions/motionVoteSubmitted/index.ts:25]: Do we want to log when that's not found?

**Review by jakubcolony — APPROVED** [review]
Nice work! 👍

File: src/handlers/motions/motionVoteSubmitted/index.ts (hunk 2 above)
Comment (willm30) [reply src/handlers/motions/motionVoteSubmitted/index.ts:25]: @jakubcolony It gets logged inside `getMotionFromDB`

**Pull request merged**